misk-api/src/main/kotlin/misk/api/RequestContext.kt

```diff
@@ -0,0 +1,16 @@
+package misk.api
+
+import misk.web.DispatchMechanism
+import okhttp3.HttpUrl
+
+interface RequestContext {
```

Comment:
This new interface (implemented by `HttpCall`) will be used for existing code in cash libraries which only relies on a bunch of request attributes. This lets us incrementally implement the compat layer, but also decouples us from the many things which are currently packed into `HttpCall`, but not used in any cash libs afaict, such as web socket-related methods.  

Given `HttpCall` implements this interface, I've created a new action scope entry for `RequestContext` with the existing  `HttpCall` instance, which means code relying on `ActionScoped<HttpCall>`, but only using some request fields, can safely use `ActionScoped<RequestContext>` instead.